mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java

```diff
@@ -121,99 +124,103 @@ protected void startUp() throws Exception {
     protected void shutDown() throws Exception {
     }
   }
+    private static class HighAvailabilityServicesImpl extends AbstractIdleService implements
+        HighAvailabilityServices {
+
+        private final MasterMonitor masterMonitor;
+        private final Counter resourceLeaderChangeCounter;
+        private final Counter resourceLeaderAlreadyRegisteredCounter;
+        private final AtomicInteger rmConnections = new AtomicInteger(0);
+        private final CoreConfiguration configuration;
+
+        public HighAvailabilityServicesImpl(CoreConfiguration configuration) {
+            this.configuration = configuration;
+            if(configuration.getLeaderMonitorFactory() instanceof LocalLeaderFactory) {
```

Comment:
This is a backwards compatibility check, to enable the injection solution to work with current configuration files. In a major version I would think we should raise in this situation.